src/cnxml/serialize.js

```diff
@@ -142,6 +143,30 @@ function list(obj, children) {
 }
 
 
+/**
+ * Serializer for source elements.
+ */
+function source(obj, children) {
+    const parser = new DOMParser();
+    const xmlDoc = parser.parseFromString(obj.getText().trim(), 'text/xml')
+    let el = document.createElement('template')
+    el.innerHTML = xmlDoc.documentElement.outerHTML
+
+    const error = xmlDoc.getElementsByTagName('parsererror')
+    if (error.length) {
+        const attrs = {
+            class: 'invalid-xml',
+        }
+        return <div {...attrs}>
+            Error: {error[0].textContent}
+            Content: {obj.getText()}
+        </div>
+    }
+    
+    return <>{el.content.childNodes}</>
```

Comment:
Why are you returning this as a React fragment, rather than just the element?

Also, why are you returning an array of elements, when there could only ever be one?

Reply:
> Why are you returning this as a React fragment, rather than just the element?

This is done because `slate-html-serializer` runs returned elements through React's `cloneElement` function, which corrupts non-react elements.